src/dateparser/dateparser.js

```diff
@@ -82,6 +108,7 @@ angular.module('ui.bootstrap.dateparser', [])
     }
 
     format = $locale.DATETIME_FORMATS[format] || format;
+    format = format.replace(SPECIAL_CHARACTERS_REGEXP, '\\$&');
```

Comment:
Hm yea, I'm referring to replacing the characters right before the parentheses are added.

With the less weird characters, I'm referring to that if given a format of `hh:mm`, the `createParser` will now receive `hh\:mm`as the format when you do this here. It doesn't need that extra `\` when looking for `hh` or `mm` (in the `angular.forEach(formatCodeToRegex…` block), it only needs the extra `\` right before creating the `regex` property when feeding it into the `RegExp` constructor with the parentheses.